src/hyperdeck/hyperdeck-core.js

```diff
@@ -12,10 +12,34 @@ var logger = Logger.get('hyperdeck.HyperdeckCore');
  * Allows you to make requests to the hyperdeck and get its parsed responses.
  * This chains the requests so only one is sent at a time.
  * You can also listen for asynchronous events sent from the hyperdeck.
- * @param ip, The IP address of the hyperdeck.
+ * @param config hyperdeck configuration
+ *               config = 'ip address string[string]'
+ *               or
+ *               config = {
+ *                 ip: 'ip address string[string]',
+ *                 [ pingInterval: ping interval in miliseconds[int][default = 10000] ],
+ *                 [ timeout: inactivity timeout in miliseconds[int][default = 3000] ]
+ *               }
  **/
-function HyperdeckCore(ip) {
+function HyperdeckCore(config) {
+  if (config === null) {
+    throw 'Invalid Configuration, please refer to documentations.';
```

Comment:
It's better to throw an Error object so there's a stack trace. 

throw new Error('...');

Reply:
```
  if (config == null || ((typeof config) !== 'string' && config.hasOwnProperty('ip') === false)) {
    throw new Error('Invalid Configuration, please refer to documentations.');
  }
```

Thinking like this

Reply:
== null is OK because it will also catch 'undefined'

But I think it would be clearer though to be more explicit like 

function isConfigValid(config) {
  return (
    typeof config === 'string' ||
    (typeof config === 'object' && !!config.ip)
  );
}

...


if (!isConfigValid(config))

Reply:
On phone and it's removed all the indentation

Reply:
if `config = null` then typeof `config === 'object'` will be true and `config.ip` will throw exception

Reply:
Ah l see in which case 

(typeof config === 'object' && && config != null && !!config.ip)

Reply:
if use `config != null` then `npm test` fails so did like below:

```
  /**
   * validate configuration
   * 
   * @param {*} config 
   */
  function isConfigValid(config) {
    return (
      config !== undefined &&
      config !== null &&
      (typeof config === 'string' || !!config.ip)
    );
  }

  // check for valid configuration
  if (!isConfigValid(config)) {
    throw new Error('Invalid Configuration, please refer to documentations.');
  }
```